cirq/ops/controlled_gate.py

```diff
@@ -86,109 +75,81 @@ def __init__(
-                    'Specified control qubit dimension <{!r}> does not match '
-                    'control_qid_shape <{!r}>.'.format(q,
-                                                       self.control_qid_shape))
+        for i, (val, dimension) in enumerate(
+                zip(self.control_values, self.control_qid_shape)):
             if not all(0 <= v < dimension for v in val):
                 raise ValueError(
-                    'Control values <{!r}> outside of range for qubit '
-                    '<{!r}>.'.format(val, q))
+                    'Control values <{!r}> outside of range for control qubit '
+                    'number <{!r}>.'.format(val, i))
 
         # Flatten nested ControlledGates.
         if isinstance(sub_gate, ControlledGate):
             self.sub_gate = sub_gate.sub_gate  # type: ignore
-            self.control_qubits += sub_gate.control_qubits
+            self.num_controls += sub_gate.num_controls
             self.control_values += sub_gate.control_values
             self.control_qid_shape += sub_gate.control_qid_shape
         else:
             self.sub_gate = sub_gate
 
-    def num_controls(self) -> int:
-        return len(self.control_qubits)
-
     def _qid_shape_(self) -> Tuple[int, ...]:
         return self.control_qid_shape + cirq.qid_shape(self.sub_gate)
 
     def _decompose_(self, qubits):
-        result = protocols.decompose_once_with_qubits(self.sub_gate,
-                                            qubits[self.num_controls():],
-                                            NotImplemented)
+        result = protocols.decompose_once_with_qubits(
+            self.sub_gate, qubits[self.num_controls:], NotImplemented)
 
         if result is NotImplemented:
             return NotImplemented
 
         decomposed = []
         for op in result:
             decomposed.append(
-                cop.ControlledOperation(qubits[:self.num_controls()], op,
+                cop.ControlledOperation(qubits[:self.num_controls], op,
                                         self.control_values))
         return decomposed
 
     def validate_args(self, qubits) -> None:
-        if len(qubits) < self.control_qubits.count(None):
-            raise ValueError('Not all control qubits specified.')
-        iter_qubits = iter(qubits)
-        for control, dimension in zip(self.control_qubits,
-                                      self.control_qid_shape):
-            if control is not None:
-                continue
-            q = next(iter_qubits)
+        if len(qubits) < self.num_qubits():
+            raise ValueError('Not all qubits specified.')
+        for q, dimension in zip(qubits, self.control_qid_shape):
             if q.dimension != dimension:
                 raise ValueError(
                     "Control qids have dimensions that don't match this gate's "
                     "control_qid_shape <{!r}>.  Qids are <{!r}>".format(
                         self.control_qid_shape, qubits))
```

Comment:
The above code can probably be replaced with `super().validate_args(qubits)`.

Reply:
Removed validate_args altogether.